File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float n;

//n is a positive float
    do
    {
        n = get_float("change owed: ");
    }
    while (0 > n);

//gets number of cents
    int change = round(n * 100);
    int coins = 0;

//determines number of quarters
    while (change >= 25)
    {
        coins++;
        change -= 25;
    }

//determines number of dimes
    while (change >= 10)
    {
        coins++;
        change -= 10;
    }

//determines number of nickels
    while (change >= 5)
    {
        coins++;
        change -= 5;
    }

//determines number of pennies
    while (change >= 1)
    {
        coins++;
        change -= 1;
    }
    printf("%i", coins);
    printf("\n");
}

File: pset6/cash/cash.py
import sys
from cs50 import get_float


def main():

    # prompt user for amount owed
    while True:
        n = get_float("change owed: ")
        if n > 0:
            break

    # calculate minimum number of coins
    n = n * 100
    coins = 0
    while n >= 25:
        n -= 25
        coins += 1
    while n >= 10:
        n -= 10
        coins += 1
    while n >= 5:
        n -= 5
        coins += 1
    while n >= 1:
        n -= 1
        coins += 1

    # print number of coins
    print(coins)


# calls main
if __name__ == "__main__":
    main()